src/schemaVisitor.ts

```diff
@@ -0,0 +1,464 @@
+import {
+  GraphQLArgument,
+  GraphQLDirective,
+  GraphQLEnumType,
+  GraphQLEnumValue,
+  GraphQLField,
+  GraphQLInputField,
+  GraphQLInputObjectType,
+  GraphQLInterfaceType,
+  GraphQLNamedType,
+  GraphQLObjectType,
+  GraphQLScalarType,
+  GraphQLSchema,
+  GraphQLUnionType,
+  Kind,
+  ValueNode,
+} from 'graphql';
+
+import {
+  getArgumentValues,
+} from 'graphql/execution/values';
+
+export type VisitableSchemaType =
+    GraphQLSchema
+  | GraphQLObjectType
+  | GraphQLInterfaceType
+  | GraphQLNamedType
+  | GraphQLScalarType
+  | GraphQLField<any, any>
+  | GraphQLArgument
+  | GraphQLUnionType
+  | GraphQLEnumType
+  | GraphQLEnumValue;
```

Comment:
There was no meaningful supertype of this hodgepodge of types that `SchemaVisitor` can visit, hence the union.